### DISCUSSION TITLE
TH16 whit Si7021 Tasmota Temp/Humidity problems

### DISCUSSION THREAD

**Discussion by darkboro88** [arendst/Tasmota#19174] [Q&A]
I have the following problem with a Sonoff TH16 with a flashed Tasmota and Si7021 sensor, and as long as I have the TH16 open on the web interface, it shows the correct values. As soon as I'm no longer on the web interface, it shows either zero values or 100% 12.6 degrees.

As soon as I open the page from the devices, the values match again, I have the problem with 2 devices behaves exactly the same.

What could be the reason? Did I miss something in the configs?

> 15:22:21.144 MQT: tele/tasmota_7FDADF/STATE = {"Time":"2023-07-23T15:22:21","Uptime":"0T00:25:50","UptimeSec":1550,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"","BSSId":":D1:35:0D","Channel":1,"Mode":"11n","RSSI":90,"Signal":-55,"LinkCount":1,"Downtime":"0T00:00:03"}}
> 15:22:21.152 MQT: tele/tasmota_7FDADF/SENSOR = {"Time":"2023-07-23T15:22:21","SI7021":{"Temperature":25.1,"Humidity":56.4,"DewPoint":15.8},"TempUnit":"C"}
> 
> 15:23:21.137 MQT: tele/tasmota_7FDADF/STATE = {"Time":"2023-07-23T15:23:21","Uptime":"0T00:26:50","UptimeSec":1610,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"","BSSId":":D1:35:0D","Channel":1,"Mode":"11n","RSSI":88,"Signal":-56,"LinkCount":1,"Downtime":"0T00:00:03"}}
> 15:23:21.144 MQT: tele/tasmota_7FDADF/SENSOR = {"Time":"2023-07-23T15:23:21","SI7021":{"Temperature":12.6,"Humidity":100.0,"DewPoint":12.6},"TempUnit":"C"}
> 
> 15:24:21.151 MQT: tele/tasmota_7FDADF/STATE = {"Time":"2023-07-23T15:24:21","Uptime":"0T00:27:50","UptimeSec":1670,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"x","BSSId":"D1:35:0D","Channel":1,"Mode":"11n","RSSI":88,"Signal":-56,"LinkCount":1,"Downtime":"0T00:00:03"}}
> 15:24:21.159 MQT: tele/tasmota_7FDADF/SENSOR = {"Time":"2023-07-23T15:24:21","SI7021":{"Temperature":12.6,"Humidity":100.0,"DewPoint":12.6},"TempUnit":"C"}
> 
> 15:25:21.126 MQT: tele/tasmota_7FDADF/STATE = {"Time":"2023-07-23T15:25:21","Uptime":"0T00:28:50","UptimeSec":1730,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"","BSSId":"D1:35:0D","Channel":1,"Mode":"11n","RSSI":92,"Signal":-54,"LinkCount":1,"Downtime":"0T00:00:03"}}
> 15:25:21.134 MQT: tele/tasmota_7FDADF/SENSOR = {"Time":"2023-07-23T15:25:21","SI7021":{"Temperature":25.2,"Humidity":56.3,"DewPoint":15.9},"TempUnit":"C"}ature":25.2,"Humidity":56.3,"DewPoint":15.9},"TempUnit":"C"}

**Comment by hcroij** [comment]
Is there any followup or solution to this problem?
I got the same.
As an extra to the post above.
I have 2 sh10 and 3 sensors (2 brand new).
1 SH has tasmota 12.5 the other 13.1, tried every possible combinations with sensor and switch.

for the rest same problem.

**Reply by darkboro88** [reply]
Hello hcroij,

I tried a few things to fix it. But what helped me in the end was that I downgraded everything to version 12.0.2. I haven't had any failures since then.

**Comment by hcroij** [comment]
Thx, 

Gone try it.

Ill keep you informed thx,.

Regards
Henk
